config.cc

```diff
@@ -24,6 +24,12 @@
 
 #if QT_VERSION >= QT_VERSION_CHECK( 5, 0, 0 )
 #include <QStandardPaths>
+  #ifdef HAVE_X11
+// Whether to XDG Base Directory specification might be followed.
+// Only Qt5 builds are supported, as Qt4 doesn't provide all functions needed
+// to get XDG Base Directory compliant locations.
+#define XDG_BASE_DIRECTORY_COMPLIANCE
+  #endif
```

Comment:
1. Grammar: "Whether to" => "Whether" ("whether to" would have to be followed by an infinitive). 
2. The indentation is weird. Just unindent everything like the surrounding code does.
3. Nesting preprocessor directives are more difficult to follow. I believe moving the new block below and checking `#if defined( HAVE_X11 ) && QT_VERSION >= QT_VERSION_CHECK( 5, 0, 0 )` would be much more readable. Note that `HAVE_X11` is usually checked before the Qt version in GoldenDict, because it is more interesting/relevant. And don't forget to separate this block with blank lines from the surrounding code.